### PR TITLE
feat: Head Cell Menu

### PR DESCRIPTION
**Pull request by a-m-dev**
PR for `<HeadCellMenu />` component.

This PR is makes us available to use that component for both SNT and PVT. 

**What is included:** 
- [x] component basic logic
- [x] flag based features for (Sorting, searching, selection and column size adjustment)
- [x] including mui 5, nebula stardust, sprout and marking them as external to be excluded in bundle
- [x] mui theming default type (`DefaultTheme`)
- [x] extracted `getMenuItemGroups` into it's own module to prepare it for testing 
- [x] and a few constant variables 

**What is not included:**
- [x] Translations (next PR to keep this one from growing) https://github.com/qlik-oss/nebula-table-utils/pull/25
- [x] unit tests (next PR to keep this one from growing) https://github.com/qlik-oss/nebula-table-utils/pull/22

###  Please Note:
The big chunk of this PR is related to pnpm-lock.yaml file (~1,100 lines) because of added packages

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -16,7 +16,17 @@ const EXCLUDED_FILES = [/node_modules/, 'src/**/__tests__', 'src/test'];
 const ENABLE_SOURCE_MAP = process.env.BUILD_SOURCE_MAPS === 'true' || process.env.NODE_ENV === 'development';
 
 const baseConfig = defineConfig({
-  external: ['react', 'react-dom', '@nebula.js/stardust'],
+  external: [
+    "react", 
+    "react-dom", 
+    "@nebula.js/stardust",
```
Comment (cbt1) [inline rollup.config.js:22]: `@nebula.js/stardust` is twice in the array

File: src/components/HeadCellMenu/HeadCellMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    <HeadCellMenuWrapper rightAligned={headTextAlign === 'right'}>
+      <StyledMenuButton
+        size="small"
+        tabIndex={tabIndex}
+        id="sn-table-head-menu-button"
+        aria-controls={openMenuDropdown ? 'sn-table-head-menu' : undefined}
+        aria-expanded={openMenuDropdown ? 'true' : undefined}
+        aria-haspopup="true"
+        onClick={handleOpenDropdown}
+        aria-label={translator.get('SNTable.Accessibility.ColumnOptions')}
```
Comment (cbt1) [inline src/components/HeadCellMenu/HeadCellMenu.tsx:82]: It might be safer to just comment this out for now until we have translator support?

File: src/components/HeadCellMenu/HeadCellMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    if (!openMenuDropdown) resetSelectionActionsEnabledStatus();
+  }, [openMenuDropdown, resetSelectionActionsEnabledStatus]);
+
+  return (
+    <HeadCellMenuWrapper rightAligned={headTextAlign === 'right'}>
+      <StyledMenuButton
+        size="small"
+        tabIndex={tabIndex}
+        id="sn-table-head-menu-button"
+        aria-controls={openMenuDropdown ? 'sn-table-head-menu' : undefined}
```
Comment (cbt1) [inline src/components/HeadCellMenu/HeadCellMenu.tsx:78]: Not really sure what we want to do with `sn-table` prefixes. But my first thought is that we should use something that is either generic (works for both charts) or explicit that is defined (ex. set as a property of the component).

File: src/components/HeadCellMenu/MenuList/MenuGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  StyledMenuItem,
+  StyledListItemIcon,
+  StyledMenuItemLabel,
+  StyledGroupLabel,
+  StyledGreenBorder,
+} from '../styles';
+import RecursiveMenuList from './RecursiveMenuList';
+
+export const interceptClickOnMenuItems = (menuGroups: MenuItemGroup[], cache: SubMenusOpenStatusCache) => {
+  const result = menuGroups.map((group) =>
```
Comment (cbt1) [inline src/components/HeadCellMenu/MenuList/MenuGroup.tsx:16]: Trying to understand what this actually is. It's gets somewhat confusing when you go from `menuGroups` into a `group` that is also a collection of groups?

File: package.json [hunk 5]
```diff
@@ -58,6 +58,7 @@
     "@testing-library/react": "^14.0.0",
     "@testing-library/user-event": "^14.4.3",
     "@types/jest": "^29.5.3",
+    "@types/qlik-engineapi": "^12.67.13",
```
Comment (cbt1) [inline package.json:61]: Why did you need to add this? Was it not update to date in `@nebula.js/stardust`?

**Comment by cbt1** [comment]
I see a lot of code where the word `column` is being used. I understand that is likely a thing coming in from the sn-table. But since a pivot table have both headers in both columns and rows. It probably makes more sense to call it something a bit generic like `Header`/`header` instead.

**Comment by a-m-dev** [comment]
> I see a lot of code where the word `column` is being used. I understand that is likely a thing coming in from the sn-table. But since a pivot table have both headers in both columns and rows. It probably makes more sense to call it something a bit generic like `Header`/`header` instead.

Actually i had a [comment somewhere in PVT PR](https://github.com/qlik-oss/sn-pivot-table/blob/30228b6284825730087cb94343520a052797930e/src/pivot-table/components/cells/DimensionTitleCell.tsx#L63) about this as well in the code. The idea is to have a much more unified way of defining `header/column` data that would make sense for both SNT and PVT, So if header Data works then that might be it ✅

File: src/components/HeadCellMenu/MenuList/MenuGroup.tsx (hunk 4 above)
Comment (a-m-dev) [reply src/components/HeadCellMenu/MenuList/MenuGroup.tsx:16]: will try to rename properly as i traverse the array

File: src/components/HeadCellMenu/HeadCellMenu.tsx (hunk 3 above)
Comment (a-m-dev) [reply src/components/HeadCellMenu/HeadCellMenu.tsx:78]: > Not really sure what we want to do with `sn-table` prefixes.

**in case of translations:**
I think [this is the solution here](https://github.com/qlik-oss/nebula-table-utils/blob/1132b21f2e6be595dd13946bf1f320240ac0d644/src/locale/locales/en.json) for us in this case, My basic idea is, if some string ends up in this repo, definitely will have it's own translation originated from this repo. 

**in case of selector identifiers** (like this one you mentioned):
we should have all selectors originated from here as well and i'll fix it soon

File: src/components/HeadCellMenu/HeadCellMenu.tsx (hunk 2 above)
Comment (a-m-dev) [reply src/components/HeadCellMenu/HeadCellMenu.tsx:82]: Yes, of course
it's on the way here: https://github.com/qlik-oss/nebula-table-utils/blob/1132b21f2e6be595dd13946bf1f320240ac0d644/src/components/HeadCellMenu/HeadCellMenu.tsx#L84

File: package.json (hunk 5 above)
Comment (a-m-dev) [reply package.json:61]: seems like that was required for type support on: 
1. `EngineAPI.IGenericHyperCubeLayout`
2. `EngineAPI.IApp`
3. `EngineAPI.IGenericObject`
4. `EngineAPI.IField`

which i need them for typing

Comment (cbt1) [reply package.json:61]: Looks like an issue with `pnpm` to me. It does not install `@types/qlik-engineapi` for some reason even tho it's a dependency in `@nebula.js/stardust` (which is installed..)

Comment (cbt1) [reply package.json:61]: I think we can ignore it for now. Also we should "pin" all versions in another PR

File: src/components/HeadCellMenu/MenuList/MenuGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+import {
+  StyledMenuItem,
+  StyledListItemIcon,
+  StyledMenuItemLabel,
+  StyledGroupLabel,
+  StyledGreenBorder,
+} from '../styles';
+import RecursiveMenuList from './RecursiveMenuList';
+
+export const interceptClickOnMenuItems = (menuSections: MenuItemGroup[], cache: SubMenusOpenStatusCache) =>
```
Comment (cbt1) [inline src/components/HeadCellMenu/MenuList/MenuGroup.tsx:15]: 👍

**Review by cbt1 — APPROVED** [review]
LGTM
